src/delta.rs

```diff
@@ -193,6 +193,14 @@ where
 
 /// Should a handle_* function be called on this element?
 fn should_handle(state: &State, config: &Config) -> bool {
+    // If interactive.diffFilter is enabled (color-only=true),
+    // output must be same as input, and must not be handled.
+    if config.color_only && (
+        *state == State::FileMeta ||
+        *state == State::CommitMeta
+    ) {
+        return false;
+    }
```

Comment:
In any cases the state is FileMeta or CommitMeta when interactiveFilter, I confirmed it never should handle. However, when some big function will be added in handle_generic_file_meta_header_line or handle_commit_meta_header_line in future, this workaround should move in each function.

Reply:
I believe that if we override options in `set.rs` this conditional logic will not be needed, right?

Reply:
Yes, it won't be needed anymore.

Reply:
Fixed here 8a5fb65